src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlMathTranslator.cs

```diff
@@ -19,117 +20,217 @@
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Cos), new[] { typeof(double) }), "COS" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sin), new[] { typeof(double) }), "SIN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Tan), new[] { typeof(double) }), "TAN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(decimal) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(double) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(float) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(int) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(long) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(sbyte) }), "SIGN" },
-            { typeof(Math).GetRuntimeMethod(nameof(Math.Sign), new[] { typeof(short) }), "SIGN" }
-        };
-
-        static readonly IEnumerable<MethodInfo> _truncateMethodInfos = new[]
-        {
-            typeof(Math).GetRuntimeMethod(nameof(Math.Truncate), new[] { typeof(decimal) }),
-            typeof(Math).GetRuntimeMethod(nameof(Math.Truncate), new[] { typeof(double) })
-        };
+        /// <summary>
+        /// The backend version to target.
+        /// </summary>
+        [CanBeNull] readonly Version _postgresVersion;
 
-        static readonly IEnumerable<MethodInfo> _roundMethodInfos = new[]
-        {
-            typeof(Math).GetRuntimeMethod(nameof(Math.Round), new[] { typeof(decimal) }),
-            typeof(Math).GetRuntimeMethod(nameof(Math.Round), new[] { typeof(double) }),
-            typeof(Math).GetRuntimeMethod(nameof(Math.Round), new[] { typeof(decimal), typeof(int) }),
-            typeof(Math).GetRuntimeMethod(nameof(Math.Round), new[] { typeof(double), typeof(int) })
-        };
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NpgsqlMathTranslator"/> class.
+        /// </summary>
+        /// <param name="postgresVersion">The backend version to target.</param>
+        public NpgsqlMathTranslator([CanBeNull] Version postgresVersion) => _postgresVersion = postgresVersion;
 
-        public virtual Expression Translate(MethodCallExpression methodCallExpression)
+        /// <inheritdoc />
+        [CanBeNull]
+        public Expression Translate(MethodCallExpression expression)
         {
-            Check.NotNull(methodCallExpression, nameof(methodCallExpression));
+            var method = expression.Method;
 
-            var method = methodCallExpression.Method;
-            if (_supportedMethodTranslations.TryGetValue(method, out string sqlFunctionName))
+            if (!method.IsStatic || method.DeclaringType != typeof(Math))
+                return null;
+
+            switch (method.Name)
```

Comment:
If we're really going with string-based method matching (not sure it's an improvement from the previous way), you can have a single case for all functions where we just upper-case the function name. Most of the C# functions correspond as-is to the PostgreSQL ones.